keps/sig-auth/3926-handling-undecryptable-resources/README.md

```diff
@@ -0,0 +1,952 @@
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [ ] Events
+  - Event Reason:
+- [ ] API .status
+  - Condition name:
+  - Other field:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [ ] Metrics
+  - Metric name:
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+<!--
+Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+implementation difficulties, etc.).
+-->
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### Does this feature depend on any specific services running in the cluster?
+
+<!--
+Think about both cluster-level services (e.g. metrics-server) as well
+as node-level agents (e.g. specific version of CRI). Focus on external or
+optional services that are needed. For example, if this feature depends on
+a cloud provider API, or upon an external software-defined storage or network
+control plane.
+
+For each of these, fill in the following—thinking about running existing user workloads
+and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]
+    - Usage description:
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+-->
+
+### Scalability
```

Comment:
@stlaz can you take a stab at answering the scalability questions already at this stage? Especially given the risks this particular feature posses?

Reply:
As I understand it, scalability usually refers to deploying the feature at scale. However, the ability to delete a resource does not change the platform scalability. I can briefly describe the scalability of the error as David pointed out in https://github.com/kubernetes/enhancements/pull/3927#discussion_r1345915347

Reply:
I mean more about the scalability impact overall, not only focusing on the feature. Imagine a cluster with hundreds of nodes, where the this feature is enabled, and then look at the questions below and see if you can answer them in a reasonable fashion. It doesn't have to be perfect, it's not a required step for alpha, and based on the initial testing at alpha you'll be able to expand on it when promoting to beta :smile:

Reply:
For example, David's question is a good way to answer the question: ` Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?` and limiting the number of values you handle at once is a potential safe guard to ensure that CPU and/or RAM isn't increased.

Reply:
A different question from that set is about SLI/SLO, is it possible that the extra operations required to force delete all the resources will affect the guaranteed time to delete a resource, will it be negligible or the expected increase is significant, etc. I hope these examples will help :wink:

Reply:
Thank you for the examples, that's quite useful :slightly_smiling_face:
 I actually addressed the original comment in a rather concise manner before I saw your additional comments but I think that it might still be a reasonable answer.